packages/hop-node/src/vault/Vault.ts

```
@@ -0,0 +1,16 @@
+import { Chain } from 'src/constants'
+import { DefaultVault } from './DefaultVault'
+import { VaultInterface } from './VaultInterface'
+import { YearnVault } from './YearnVault'
+
+export class Vault extends DefaultVault implements VaultInterface {
+  static from (chain: Chain, token: string, signer: any): VaultInterface {
+    const yearnVaultTokens = new Set(['ETH', 'USDC', 'USDT', 'DAI'])
+    const useYearnVault = chain === Chain.Ethereum && yearnVaultTokens.has(token)
+    if (useYearnVault) {
+      return new YearnVault(chain, token, signer)
+    }
+
+    return new DefaultVault(chain, token, signer)
```

Returning a non-functional vault when none exists seems counterintuitive. I'd expect it to throw or return undefined to be handled downstream.

---

If you agree with the above, I'd consider removing `DefaultVault` and `VaultInterface` entirely and then change this class to an abstract class with all the functions in `VaultInterface` but also abstract alongside the `from` static function. That way there's just a single base class that has the `from` helper for creating new Vaults that inherit from it.

There might be a downside I'm missing and I think the current structure works well as is too.

---

Good suggestion, the redundant classes/interfaces have been removed and updated to be abstract class

Resolved in https://github.com/hop-protocol/hop/pull/404/commits/bcd375614190d12e4aecaa4891c4d8715acafac8